Automattic/pocket-casts-android
# Sync playback effects setting

## Description

One of the settings listed for sync project #1739. This PR adds syncing to the global playback effects. This setting can be changed in multiple ways: Phone app, Notification center, WearOS app, Tasker app.

## Testing prerequisites

Each test requires two devices to check syncing. Sync process looks as follows.

1. Make sure that both devices are synced before starting the test by refreshing apps in the profile.
2. Change the setting on the device 1.
3. Refresh settings on the device 1.
4. Refresh settings on the device 2.
5. Check settings on the device 2.

### Phone app

Settings configuration can be reached by playing a podcast, opening the player, and then opening playback effects. You need to make sure that those are global settings. If settings are local to podcast you'll see a banner at the bottom that allows to `Clear` the settings.

![Screenshot_20240126-125633](https://github.com/Automattic/pocket-casts-android/assets/30936061/991e251a-7b7c-4a97-9fbc-98c9bf79eca5)

### Notification center

You can change the playback speed from the notification center. Play a podcast and open notification center. I'm not sure since which version it is available. I tested with Android 14.

![Screenshot_20240126-125622](https://github.com/Automattic/pocket-casts-android/assets/30936061/da8d358c-e4a2-4f32-8ddc-6798ade69e3b)

### WearOS app

If you have a physical device you should be able to test it without any issues. If you don't you can test it with an emulator but it'll require some work.

Build the *release* version of the app with this condition inverted.

```diff
--- a/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/featureflag/Feature.kt
+++ b/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/featureflag/Feature.kt
@@ -30,7 +30,7 @@ enum class Feature(
     SETTINGS_SYNC(
         key = "settings_sync",
         title = "Settings Sync",
-        defaultValue = BuildConfig.DEBUG,
+        defaultValue = !BuildConfig.DEBUG,
         tier = FeatureTier.Free,
         hasFirebaseRemoteFlag = false,
         hasDevToggle = true,

```

1. Have a phone with non A8C account (managed profiles don't work with emulators). This account should have also a Pocket Casts Plus account.
2. Create a wearOS emulator and pair it with the phone.
3. Install **release** versions of the app on both the phone and the wear device.
4. Sign in to the Pocket Casts app using Google sign in. You should be ready to test now.

To be able to control playback effects from the device you need to play an episode. After that you need to enter the player screen where you'll see a playback effects icon.

| Player | Effects | Settings sync |
| - | - | - |
| ![Screenshot_20240126_121809](https://github.com/Automattic/pocket-casts-android/assets/30936061/2df43655-b6a2-4a1d-8902-65e5aa90b688) | ![Screenshot_20240126_121840](https://github.com/Automattic/pocket-casts-android/assets/30936061/e0d6ef2b-16e1-47d5-b4c9-f3719dd4d787) | ![Screenshot_20240126_121854](https://github.com/Automattic/pocket-casts-android/assets/30936061/97026409-3d81-4590-aa00-e21da35da0f8) |

### Tasker app

This is an external app that you can expense and download from the [Play Store](https://play.google.com/store/apps/details?id=net.dinglisch.android.taskerm). We support it via plugin. Once you have the app create the following tasks and play them (you can use different values).

![Screenshot_20240126-125646](https://github.com/Automattic/pocket-casts-android/assets/30936061/16e4ced6-d1ed-40c1-a4a5-aeb5c89258a0)

## Testing Instructions

I won't provide step by step testing instructions. Basically test that playback effects sync correctly using different methods that I described. Test each setting separately, combinations of some, and all of them. If you want a more comprehensive guide on how to test syncing check, for example, [this PR](https://github.com/Automattic/pocket-casts-android/pull/1736).

I found a bug with the notifications center not displaying speeds below `1x` - #1745.

## Checklist
 
#### I have tested any UI changes...
